pkg/virt-handler/vm.go

```diff
@@ -500,46 +528,67 @@ func (d *VirtualMachineController) getLauncherClient(vmi *v1.VirtualMachineInsta
 
 func (d *VirtualMachineController) processVmShutdown(vmi *v1.VirtualMachineInstance, domain *api.Domain) error {
 
-	clientDisconnected := false
+	// Only attempt to shutdown/destroy if we still have a connection established with the pod.
+	client, err := d.getVerifiedLauncherClient(vmi)
 
-	client, err := d.getLauncherClient(vmi)
-	if err != nil {
-		clientDisconnected = true
-	}
+	// If the pod has been torn down, we know the VirtualMachineInstance is down.
+	if err == nil {
```

Comment:
Can you do `if err != nil` then return `err` here?

Reply:
yes